lib/rules/declaration-empty-line-before/index.mjs

```diff
@@ -127,22 +128,27 @@ const rule = (primary, secondaryOptions, context) => {
 				return;
 			}
 
-			// Fix
-			if (context.fix) {
-				if (context.newline == null) return;
-
-				if (expectEmptyLineBefore) {
-					addEmptyLineBefore(decl, context.newline);
-				} else {
-					removeEmptyLinesBefore(decl, context.newline);
-				}
-
-				return;
-			}
-
 			const message = expectEmptyLineBefore ? messages.expected : messages.rejected;
+			const mutate = expectEmptyLineBefore ? addEmptyLineBefore : removeEmptyLinesBefore;
+			const semicolon = decl.parent?.raws.semicolon ? 1 : 0;
+			const endIndex = decl.toString().length + semicolon;
```

Comment:
major

this is an incorrect assumption
i.e. it fails for `a { bottom: 12px; top: 17px }`